### DISCUSSION TITLE
adhoc change layers of existing model

### DISCUSSION THREAD

**Discussion by LeonhardFeiner** [google/flax#2248] [Q&A] [answered]
In TensorFlow, there exists `tf.keras.models.clone_model` which together with the `clone_function` can be used to ad-hoc change/replace layers of an existing model object.
https://www.tensorflow.org/api_docs/python/tf/keras/models/clone_model

Do you have an idea, how I could generically replace every e.g. conv layer in a model?

I looked into the implementation and in my opinion the model's submodules are initialized mostly on the fly while calling `nn.compact` and therefore don't exist in the model object at all before calling / compiling the model.

**Comment by cgarciae** [comment, marked as answer]
Hey @LeonhardFeiner, as you point out there is no good way to do this, even `setup` is called lazily so you don't even have access to sub-modules defined there from the outside. This probably means that for now you probably have to fork the whole model architecture and make changes yourself.
